### PR TITLE
Fix for going back taking you to the incorrect place during registration

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this could use extra scrutiny from @chromakode and @MarcoPolo 
I'm fixing the case where you (on mobile)

1. Start clean (no users)
1. Try and provision as a device (click 'log in')
1. Click back button

It takes you to the loginTab/login (the relogin screen) and not the root loginTab.

I think the real fix is a larger overhaul of our routes in provisioning and something like
loginTab/reLogin
loginTab/newLogin
or something. This seemed like a smaller short term fix

### CODE REVIEW

File: shared/actions/login/saga.js [hunk 1]
```diff
@@ -162,7 +162,12 @@ const kex2Sagas = (onBackSaga, provisionerSuccessSaga) => ({
 })
 
 function* cancelLogin() {
-  yield put(navigateTo(['login'], [loginTab]))
+  const getNumAccounts = (state: TypedState) =>
```
Comment (chrisnojima) [inline shared/actions/login/saga.js]: go to the relogin or the root page depending on if you have accounts or not

File: shared/actions/login/saga.js [hunk 2]
```diff
@@ -428,7 +433,7 @@ function* handleProvisioningError(error) {
     )
   )
   yield race({onBack: take(Constants.onBack), navUp: take(RouteConstants.navigateUp)})
-  yield put(navigateTo(['login'], [loginTab]))
+  yield call(cancelLogin)
```
Comment (chrisnojima) [inline shared/actions/login/saga.js]: just delegate this to the helper that does the right thing

File: shared/actions/login/saga.js [hunk 3]
```diff
@@ -488,7 +493,10 @@ function* initalizeMyCodeStateForAddingADevice() {
 }
 
 function* startLoginSaga() {
-  yield put(navigateAppend(['usernameOrEmail'], [loginTab, 'login']))
+  yield put(Creators.setLoginFromRevokedDevice(''))
```
Comment (chrisnojima) [inline shared/actions/login/saga.js]: the component used to dispatch all these actions but really they should be one thing. i moved this here but really we should likely just listen to the startlogin action in the reducer and clear all this stuff out

File: shared/actions/signup.js [hunk 4]
```diff
@@ -84,11 +84,12 @@ function checkInviteCode(
     })
 }
 
-function requestAutoInvite(): TypedAsyncAction<
-  CheckInviteCode | NavigateAppend | NavigateTo | SignupWaiting
```
Comment (chrisnojima) [inline shared/actions/signup.js]: i think its maybe not that useful to type the types of actions our actions will possibly dispatch. dunno how much we actually get by these guarantees.

File: shared/login/forms/intro.js [hunk 5]
```diff
@@ -24,19 +22,12 @@ const mapDispatchToProps = (dispatch: Dispatch, {navigateAppend}) => ({
     dispatch(navigateAppend(['feedback']))
   },
   onLogin: () => {
```
Comment (chrisnojima) [inline shared/login/forms/intro.js]: i think this is an anti-pattern. components shouldn't know this much

**Comment by keybase-ci-visdiff** [comment]
The commits 3a78fd4d5d23...f5b8083a1398 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497360158525-3a78fd4d5d23-f5b8083a1398-linux/Tracker-New_user_Scroll_1-3a78fd4d5d23.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1497360158525-3a78fd4d5d23-f5b8083a1398-linux/Tracker-New_user_Scroll_1-f5b8083a1398.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Pull request merged**